src/formats/apple-notes/convert-note.ts

```diff
@@ -0,0 +1,398 @@
+		let row, id, converter;
+		
+		switch (attr.attachmentInfo.typeUti) {
+			case ANAttachment.Hashtag:
+			case ANAttachment.Mention:
+				row = await this.importer.database.get`
+					SELECT zalttext FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				return row.ZALTTEXT;
+				
+			case ANAttachment.InternalLink:
+				row = await this.importer.database.get`
+					SELECT ztokencontentidentifier FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				const destIdentifier = row.ZTOKENCONTENTIDENTIFIER.match(NOTE_URI)[1];
+				
+				row = await this.importer.database.get`
+					SELECT z_pk FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${destIdentifier.toUpperCase()}`;
+					
+				if (!(row.Z_PK in this.importer.resolvedFiles)) {
+					//if we don't have the note yet, we fetch it early to get its path
+					this.importer.resolveNote(row.Z_PK);
+				}
+				
+				return this.app.fileManager.generateMarkdownLink(
+					this.importer.resolvedFiles[row.Z_PK] as TFile, 
+					this.importer.rootFolder.path
+				);
+				
+			case ANAttachment.Table:
+				row = await this.importer.database.get`
+					SELECT hex(zmergeabledata1) as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+
+				converter = this.importer.decodeData(row.zhexdata, TableConverter);
+				return await converter.format();
+			
+			case ANAttachment.UrlCard:
+				row = await this.importer.database.get`
+					SELECT ztitle, zurlstring as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+					
+				return `[**${row.ZTITLE}**](${row.ZURLSTRING})`;
+			
+			case ANAttachment.Scan:
+				row = await this.importer.database.get`
+					SELECT hex(zmergeabledata1) as zhexdata FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				converter = this.importer.decodeData(row.zhexdata, ScanConverter);
+				return await converter.format();
+			
+			case ANAttachment.ModifiedScan:
+			case ANAttachment.Drawing:
+				row = await this.importer.database.get`
+					SELECT z_pk FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				id = row?.Z_PK;
+				break;
+				
+			//actual file on disk (eg image, audio, video, pdf, vcard)
+			//hundreds of different utis so not in the enum
+			default:
+				row = await this.importer.database.get`
+					SELECT zmedia FROM ziccloudsyncingobject 
+					WHERE zidentifier = ${attr.attachmentInfo.attachmentIdentifier}`;
+				
+				id = row?.ZMEDIA;
+				break;
+		}
+		
+		if (!id) {
+			//doesn't have an associated file, so unknown
+			return ` **(unknown attachment: ${attr.attachmentInfo.typeUti})** `;
+		}
+		
+		await this.importer.resolveAttachment(id, attr.attachmentInfo.typeUti);
```

Comment:
We should make `resolveAttachment` return the TFile, so we don't have to look it up again after.